### PR TITLE
Use CCSM-HELM retention policy

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

* Use the newly introduced retention policy, which will deploy a curator config and job (similar to our deployments)
* [remove old curator manifests](https://github.com/camunda-cloud/zeebe/commit/687b1cae2cceacde79fa8afc33955ee3fd25e01a)
* [remove old values](https://github.com/camunda-cloud/zeebe/commit/082011fc4072415c480fcf573a0c4699b055bb8f)
* [remove curator manifests from Makefile](https://github.com/camunda-cloud/zeebe/commit/5c984bfcd6f40e83568b3fe16e5db388cdac93a6)

I run a benchmark with the changes.

Logs from curator:

```
$ k logs ccsm-curator-27433770-clqgb 
/opt/bitnami/python/lib/python3.9/site-packages/elasticsearch_curator-5.8.4-py3.9.egg/curator/actions.py:151: SyntaxWarning: "is" with a literal. Did you mean "=="?
/opt/bitnami/python/lib/python3.9/site-packages/elasticsearch_curator-5.8.4-py3.9.egg/curator/actions.py:151: SyntaxWarning: "is" with a literal. Did you mean "=="?
2022-02-28 05:30:03,816 INFO      Preparing Action ID: 1, "delete_indices"
2022-02-28 05:30:03,816 INFO      Creating client object and testing connection
2022-02-28 05:30:03,820 INFO      Instantiating client object
2022-02-28 05:30:03,821 INFO      Testing client connectivity
2022-02-28 05:30:03,837 INFO      Successfully created Elasticsearch client object with provided settings
2022-02-28 05:30:03,842 INFO      Trying Action ID: 1, "delete_indices": Clean up ES by deleting old Zeebe indices
2022-02-28 05:30:03,888 INFO      Skipping action "delete_indices" due to empty list: <class 'curator.exceptions.NoIndices'>
2022-02-28 05:30:03,888 INFO      Action ID: 1, "delete_indices" completed.
2022-02-28 05:30:03,888 INFO      Preparing Action ID: 2, "delete_indices"
2022-02-28 05:30:03,888 INFO      Creating client object and testing connection
2022-02-28 05:30:03,889 INFO      Instantiating client object
2022-02-28 05:30:03,889 INFO      Testing client connectivity
2022-02-28 05:30:03,901 INFO      Successfully created Elasticsearch client object with provided settings
2022-02-28 05:30:03,905 INFO      Trying Action ID: 2, "delete_indices": Clean up ES by deleting old Operate indices
2022-02-28 05:30:03,936 INFO      Skipping action "delete_indices" due to empty list: <class 'curator.exceptions.NoIndices'>
2022-02-28 05:30:03,936 INFO      Action ID: 2, "delete_indices" completed.
2022-02-28 05:30:03,937 INFO      Preparing Action ID: 3, "delete_indices"
2022-02-28 05:30:03,937 INFO      Creating client object and testing connection
2022-02-28 05:30:03,937 INFO      Instantiating client object
2022-02-28 05:30:03,938 INFO      Testing client connectivity
2022-02-28 05:30:03,948 INFO      Successfully created Elasticsearch client object with provided settings
2022-02-28 05:30:03,952 INFO      Trying Action ID: 3, "delete_indices": Clean up ES by deleting old Tasklist indices
2022-02-28 05:30:03,987 INFO      Skipping action "delete_indices" due to empty list: <class 'curator.exceptions.NoIndices'>
2022-02-28 05:30:03,988 INFO      Action ID: 3, "delete_indices" completed.
2022-02-28 05:30:03,988 INFO      Preparing Action ID: 4, "delete_indices"
2022-02-28 05:30:03,988 INFO      Creating client object and testing connection
2022-02-28 05:30:03,988 INFO      Instantiating client object
2022-02-28 05:30:03,989 INFO      Testing client connectivity
2022-02-28 05:30:03,999 INFO      Successfully created Elasticsearch client object with provided settings
2022-02-28 05:30:04,002 INFO      Trying Action ID: 4, "delete_indices": Clean up ES by deleting too big Zeebe indices
2022-02-28 05:30:04,035 INFO      Deleting 1 selected indices: ['zeebe-record_job_1.4.0-snapshot_2022-02-28']
2022-02-28 05:30:04,035 INFO      ---deleting index zeebe-record_job_1.4.0-snapshot_2022-02-28
2022-02-28 05:30:04,942 INFO      Action ID: 4, "delete_indices" completed.
2022-02-28 05:30:04,942 INFO      Job completed.
```

Benchmark metrics:

elastic
general

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

### CODE REVIEW

File: benchmarks/setup/default/zeebe-values.yaml [hunk 1]
```diff
@@ -110,6 +110,23 @@ zeebe-gateway:
...
+  # RetentionPolicy.enabled if true, elasticsearch curator cronjob and configuration will be deployed.
+  enabled: true
+  # RetentionPolicy.schedule defines how often/when the curator should run
+  schedule: "*/15 * * * *"
+  # RetentionPolicy.zeebeIndexTTL defines after how many days a zeebe index can be deleted
+  zeebeIndexTTL: 1
+  # RetentionPolicy.zeebeIndexMaxSize can be set to configure the maximum allowed zeebe index size in gigabytes.
+  # After reaching that size, curator will delete that corresponding index on the next run.
+  # To benefit from that configuration the schedule needs to be configured small enough, like every 15 minutes.
+  zeebeIndexMaxSize: 10
```
Comment (oleschoenburg) [inline benchmarks/setup/default/zeebe-values.yaml:124]: :grey_question: Does that mean that indices get deleted if they are older than one day *or* larger than 10GiB?

**Review by lenaschoenburg — APPROVED** [review]
Looks great, thanks @Zelldon :+1:

File: benchmarks/setup/default/zeebe-values.yaml (hunk 1 above)
Comment (Zelldon) [reply benchmarks/setup/default/zeebe-values.yaml:124]: Yes this is correct :)

**Pull request merged**